tests/controls/gridtest.cpp

```diff
@@ -1635,4 +2224,197 @@ TEST_CASE("GridBlockCoords::SymDifference", "[grid]")
+
+    // String s contains the rendering of the grid, start with drawing
+    // the header (more than 26 columns not properly supported).
+    wxString s(' ', colMargin + 1);
+    for ( int col = 0; col < numCols; ++col )
+        s += wxChar('A' + col);
+    s += "\n";
+
+    s += wxString(' ', colMargin) + '+' + wxString('-', numCols) + "\n";
+
+    const int absEditCount = abs(m_edit.count);
+    wxString action;
+    action.Printf(" %s: %d",
+                  m_edit.count < 0 ? "deletions" : "insertions",
+                  absEditCount);
+
+    // Will contain summary of grid (only multicells mentioned).
+    wxString content;
+
+    // Draw grid content.
+    for ( int row = 0; row < numRows; ++row )
+    {
+        const wxString label = GetRowLabelValue(row);
+        s += wxString(' ', colMargin - label.length());
+        s += label + '|';
+
+        for ( int col = 0; col < numCols; ++col )
+        {
+            wxChar c = 'x';
+            int rows, cols;
+            switch ( GetCellSize(row, col, &rows, &cols) )
+            {
+            case wxGrid::CellSpan_None:
+                c = HasAttr(row, col) ? '*' : '.';
+                break;
+
+            case wxGrid::CellSpan_Main:
+                c = 'M';
+                content += Multicell(row, col, rows, cols).ToString() + "\n";
+                break;
+
+            case wxGrid::CellSpan_Inside:
+                // Check if the offset to main cell is correct.
+                c = (GetCellSize(row + rows, col + cols, &rows, &cols)
+                        == wxGrid::CellSpan_Main) ? 'm' : '?';
+                break;
+            }
+
+            s += c;
+        }
+
+        // If applicable draw annotated row edits.
+        if ( m_edit.count && m_edit.isRow
+             && row >= m_edit.pos && row < m_edit.pos + absEditCount)
+        {
+            s += (m_edit.count < 0 ? " ^" : " v");
+            if ( row == m_edit.pos )
+                s += action;
+        }
+
+        s += "\n";
+    }
+
+    // Draw annotated footer if columns edited.
+    if ( m_edit.count && !m_edit.isRow )
+    {
+        s += wxString(' ', colMargin + 1);
+
+        for ( int col = 0; col < m_edit.pos + absEditCount; ++col )
+            if ( col < m_edit.pos )
+                s += " ";
+            else
+                s += (m_edit.count < 0 ? "<" : ">");
+
+        s += action + "\n";
+    }
+
+    s += "\n";
+
+    if ( content.empty() )
+        content = "Empty grid\n";
+
+    return content + s;
+}
+
+//
+// GridMatcher
+//
+
+GridMatcher::GridMatcher(const TestableGrid& grid) : m_grid(&grid)
+{
+   m_expectedGridDesc = m_grid->ToString();
+}
+
+bool GridMatcher::match(const TestableGrid& other) const
+{
+    const int rows = wxMax( m_grid->GetNumberRows(), other.GetNumberRows() );
+    const int cols = wxMax( m_grid->GetNumberCols(), other.GetNumberCols() );
+
+    int row, col;
+    for ( row = 0; row < rows; ++row )
+    {
+        for ( col = 0; col < cols; ++col )
```

Comment:
This is really minor, but why not
```suggestion
    for ( int row = 0; row < rows; ++row )
    {
        for ( int col = 0; col < cols; ++col )
```

Reply:
Thanks, remnants from before when the matcher was matching a grid against a `Multicell` only and there was another 2D loop checking for spurious attributes.